--- hamcrest-0.1.0/Cargo.toml
+++ hamcrest-0.1.0/Cargo.toml
@@ -5,4 +5,4 @@
 authors = ["Carllerche <me@carllerche.com>"]
 
 [dependencies]
-num = "0.1.25"
+num = { version = "0.1.25", path = "../num-0.1.25" }
--- rand-0.3.8/Cargo.toml
+++ rand-0.3.8/Cargo.toml
@@ -14,11 +14,7 @@
 keywords = ["random"]
 
 [dependencies]
-libc = "0.1.1"
-
-[dev-dependencies.rand_macros]
-path = "rand_macros"
-version = "*"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
 
 [dev-dependencies]
-log = "0.3.0"
+log = { version = "0.3.1", path = "../log-0.3.1" }
--- tempdir-0.3.4/Cargo.toml
+++ tempdir-0.3.4/Cargo.toml
@@ -14,4 +14,4 @@
 """
 
 [dependencies]
-rand = "0.3"
+rand = { version = "0.3.8", path = "../rand-0.3.8" }
--- num-0.1.25/Cargo.toml
+++ num-0.1.25/Cargo.toml
@@ -14,8 +14,8 @@
 """
 
 [dependencies]
-rustc-serialize = { version = "0.3.13", optional = true }
-rand = { version = "0.3.8", optional = true }
+rustc-serialize = { version = "0.3.14", optional = true, path = "../rustc-serialize-0.3.14" }
+rand = { version = "0.3.8", optional = true, path = "../rand-0.3.8" }
 
 [features]
 
--- toml-0.1.21/Cargo.toml
+++ toml-0.1.21/Cargo.toml
@@ -16,7 +16,7 @@
 """
 
 [dependencies]
-rustc-serialize = { optional = true, version = "0.3.0", path = "../rustc-serialize-0.3.14" }
+rustc-serialize = { optional = true, version = "0.3.14", path = "../rustc-serialize-0.3.14" }
 
 [features]
 default = ["rustc-serialize"]
--- cargo-0.4.0/Cargo.toml
+++ cargo-0.4.0/Cargo.toml
@@ -40,6 +40,12 @@
 toml = { version = "0.1.21", path = "../toml-0.1.21" }
 url = { version = "0.2.35", path = "../url-0.2.35" }
 
+[dev-dependencies]
+hamcrest = { version = "0.1.0", path = "../hamcrest-0.1.0" }
+bufstream = { version = "0.1.1", path = "../bufstream-0.1.1" }
+tempdir = { version = "0.3.4", path = "../tempdir-0.3.4" }
+filetime = { version = "0.1.5", path = "../filetime-0.1.5" } 
+
 [[bin]]
 name = "cargo"
 test = false
--- cargo-0.4.0/Makefile.in
+++ cargo-0.4.0/Makefile.in
@@ -88,7 +88,9 @@
 
 test-unit-$(1): $$(CARGO)
 	@mkdir -p target/$(1)/cit
-	$$(CARGO) test --target $(1) $$(VERBOSE_FLAG) $$(only)
+	"$$(CFG_RUSTC)" -V
+	"$$(CARGO)" --version
+	$$(CARGO) test --target $(1) $$(OPT_FLAG) $$(VERBOSE_FLAG) $$(only)
 endef
 $(foreach target,$(CFG_TARGET),$(eval $(call CARGO_TARGET,$(target))))
 
@@ -99,7 +101,7 @@
 
 # === Tests
 
-test: style no-exes $(foreach target,$(CFG_TARGET),test-unit-$(target))
+test: style $(foreach target,$(CFG_TARGET),test-unit-$(target))
 
 style:
 	sh tests/check-style.sh
