diff --git a/aho-corasick-0.3.0/Cargo.toml b/aho-corasick-0.3.0/Cargo.toml
index 50efb37..14deffd 100644
--- a/aho-corasick-0.3.0/Cargo.toml
+++ b/aho-corasick-0.3.0/Cargo.toml
@@ -20,14 +20,7 @@ doc = false
 bench = false
 
 [dependencies]
-memchr = "0.1.*"
-
-[dev-dependencies]
-csv = "*"
-docopt = "*"
-quickcheck = "*"
-rand = "*"
-rustc-serialize = "*"
+memchr = { version = "0.1.3", path = "../memchr-0.1.3" }
 
 [[bench]]
 name = "bench"
diff --git a/cargo-0.5.0/Cargo.toml b/cargo-0.5.0/Cargo.toml
index 5c360c8..ef59f76 100644
--- a/cargo-0.5.0/Cargo.toml
+++ b/cargo-0.5.0/Cargo.toml
@@ -17,37 +17,34 @@ name = "cargo"
 path = "src/cargo/lib.rs"
 
 [dependencies]
-advapi32-sys = "0.1"
-curl = "0.2"
-docopt = "0.6"
-env_logger = "0.3"
-filetime = "0.1"
-flate2 = "0.2"
-git2 = "0.2"
-git2-curl = "0.2"
-glob = "0.2"
-kernel32-sys = "0.1"
-libc = "0.1"
-libgit2-sys = "0.2"
-log = "0.3"
-num_cpus = "0.2"
-regex = "0.1"
+curl = { version = "0.2.11", path = "../curl-0.2.11" }
+docopt = { version = "0.6.70", path = "../docopt-0.6.70" }
+env_logger = { version = "0.3.1", path = "../env_logger-0.3.1" }
+filetime = { version = "0.1.5", path = "../filetime-0.1.5" }
+flate2 = { version = "0.2.7", path = "../flate2-0.2.7" }
+git2 = { version = "0.2.14", path = "../git2-0.2.14" }
+git2-curl = { version = "0.2.4", path = "../git2-curl-0.2.4" }
+glob = { version = "0.2.10", path = "../glob-0.2.10" }
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
+libgit2-sys = { version = "0.2.20", path = "../libgit2-sys-0.2.20" }
+log = { version = "0.3.1", path = "../log-0.3.1" }
+num_cpus = { version = "0.2.6", path = "../num_cpus-0.2.6" }
+regex = { version = "0.1.41", path = "../regex-0.1.41" }
 crates-io = { path = "src/crates-io", version = "0.1" }
-rustc-serialize = "0.3"
-semver = "0.1"
-tar = "0.3"
-term = "0.2"
-threadpool = "0.1"
-time = "0.1"
-toml = "0.1"
-url = "0.2"
-winapi = "0.2"
+rustc-serialize = { version = "0.3.16", path = "../rustc-serialize-0.3.16" }
+semver = { version = "0.1.20", path = "../semver-0.1.20" }
+tar = { version = "0.3.1", path = "../tar-0.3.1" }
+term = { version = "0.2.11", path = "../term-0.2.11" }
+threadpool = { version = "0.1.4", path = "../threadpool-0.1.4" }
+time = { version = "0.1.32", path = "../time-0.1.32" }
+toml = { version = "0.1.22", path = "../toml-0.1.22" }
+url = { version = "0.2.37", path = "../url-0.2.37" }
 
 [dev-dependencies]
-tempdir = "0.3"
-hamcrest = "0.1"
-bufstream = "0.1"
-filetime = "0.1"
+tempdir = { version = "0.3.4", path = "../tempdir-0.3.4" }
+hamcrest = { version = "0.1.0", path = "../hamcrest-0.1.0" }
+bufstream = { version = "0.1.1", path = "../bufstream-0.1.1" }
+filetime = { version = "0.1.5", path = "../filetime-0.1.5" }
 
 [[bin]]
 name = "cargo"
diff --git a/cargo-0.5.0/src/crates-io/Cargo.toml b/cargo-0.5.0/src/crates-io/Cargo.toml
index 70068ec..e21447d 100644
--- a/cargo-0.5.0/src/crates-io/Cargo.toml
+++ b/cargo-0.5.0/src/crates-io/Cargo.toml
@@ -13,5 +13,5 @@ name = "crates_io"
 path = "lib.rs"
 
 [dependencies]
-curl = "0.2"
-rustc-serialize = "0.3"
+curl = { version = "0.2.11", path = "../../../curl-0.2.11" }
+rustc-serialize = { version = "0.3.16", path = "../../../rustc-serialize-0.3.16" }
diff --git a/cmake-0.1.3/Cargo.toml b/cmake-0.1.3/Cargo.toml
index 44a2a19..0e36fdb 100644
--- a/cmake-0.1.3/Cargo.toml
+++ b/cmake-0.1.3/Cargo.toml
@@ -14,4 +14,4 @@ A build dependency for running `cmake` to build a native library
 """
 
 [dependencies]
-gcc = "0.3.11"
+gcc = { version = "0.3.13", path = "../gcc-0.3.13" }
diff --git a/curl-0.2.11/Cargo.toml b/curl-0.2.11/Cargo.toml
index 27429c6..9b42f57 100644
--- a/curl-0.2.11/Cargo.toml
+++ b/curl-0.2.11/Cargo.toml
@@ -8,37 +8,34 @@ repository = "https://github.com/carllerche/curl-rust"
 description = "Rust bindings to libcurl for making HTTP requests"
 
 [dependencies]
-url = "0.2.0"
-log = "0.3.0"
-libc = "0.1"
-curl-sys = { path = "curl-sys", version = "0.1.0" }
-
-[dev-dependencies]
-env_logger = "0.3.0"
+url = { version = "0.2.37", path = "../url-0.2.37" }
+log = { version = "0.3.1", path = "../log-0.3.1" }
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
+curl-sys = { path = "../curl-sys-0.1.25", version = "0.1.25" }
 
 # Unix platforms use OpenSSL for now to provide SSL functionality
 [target.i686-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.i686-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.arm-unknown-linux-gnueabihf.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.aarch64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.i686-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-bitrig.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-openbsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-dragonfly.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 
 [[test]]
 
diff --git a/curl-sys-0.1.25/Cargo.toml b/curl-sys-0.1.25/Cargo.toml
index db65050..8f42173 100644
--- a/curl-sys-0.1.25/Cargo.toml
+++ b/curl-sys-0.1.25/Cargo.toml
@@ -9,33 +9,33 @@ repository = "https://github.com/carllerche/curl-rust"
 description = "Native bindings to the libcurl library"
 
 [build-dependencies]
-pkg-config = "0.3"
-gcc = "0.3.10"
+pkg-config = { version = "0.3.5", path = "../pkg-config-0.3.5" }
+gcc = { version = "0.3.13", path = "../gcc-0.3.13" }
 
 [lib]
 name = "curl_sys"
 path = "lib.rs"
 
 [dependencies]
-libz-sys = "0.1.0"
-libc = "0.1"
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
 
 # Unix platforms use OpenSSL for now to provide SSL functionality
 [target.i686-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.i686-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.arm-unknown-linux-gnueabihf.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.aarch64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.i686-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-bitrig.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
diff --git a/docopt-0.6.70/Cargo.toml b/docopt-0.6.70/Cargo.toml
index 41e29f6..e370f28 100644
--- a/docopt-0.6.70/Cargo.toml
+++ b/docopt-0.6.70/Cargo.toml
@@ -20,6 +20,6 @@ doc = false
 test = false
 
 [dependencies]
-regex = "0.1"
-rustc-serialize = "0.3"
-strsim = "0.3"
+regex = { version = "0.1.41", path = "../regex-0.1.41" }
+rustc-serialize = { version = "0.3.16", path = "../rustc-serialize-0.3.16" }
+strsim = { version = "0.3.0", path = "../strsim-0.3.0" }
diff --git a/env_logger-0.3.1/Cargo.toml b/env_logger-0.3.1/Cargo.toml
index 57f46e5..9ea1e95 100644
--- a/env_logger-0.3.1/Cargo.toml
+++ b/env_logger-0.3.1/Cargo.toml
@@ -12,8 +12,8 @@ variable.
 """
 
 [dependencies.log]
-version = "0.3"
-path = ".."
+version = "0.3.1"
+path = "../log-0.3.1"
 
 [dependencies]
-regex = "0.1"
+regex = { version = "0.1.41", path = "../regex-0.1.41" }
diff --git a/filetime-0.1.5/Cargo.toml b/filetime-0.1.5/Cargo.toml
index ccbf114..9b0ca20 100644
--- a/filetime-0.1.5/Cargo.toml
+++ b/filetime-0.1.5/Cargo.toml
@@ -13,9 +13,4 @@ Platform-agnostic accessors of timestamps in File metadata
 """
 
 [dependencies]
-libc = "0.1"
-kernel32-sys = "0.1"
-winapi = "0.2"
-
-[dev-dependencies]
-tempdir = "0.3"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
diff --git a/filetime-0.1.5/src/lib.rs b/filetime-0.1.5/src/lib.rs
index 979367a..acac6b2 100644
--- a/filetime-0.1.5/src/lib.rs
+++ b/filetime-0.1.5/src/lib.rs
@@ -35,7 +35,9 @@
 //! ```
 
 extern crate libc;
+#[cfg(windows)]
 extern crate winapi;
+#[cfg(windows)]
 extern crate kernel32;
 
 #[cfg(unix)] use std::os::unix::prelude::*;
diff --git a/flate2-0.2.7/Cargo.toml b/flate2-0.2.7/Cargo.toml
index 608608d..23422b5 100644
--- a/flate2-0.2.7/Cargo.toml
+++ b/flate2-0.2.7/Cargo.toml
@@ -16,8 +16,5 @@ streams.
 """
 
 [dependencies]
-libc = "0.1"
-miniz-sys = { path = "miniz-sys", version = "0.1" }
-
-[dev-dependencies]
-rand = "0.3"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
+miniz-sys = { path = "../miniz-sys-0.1.5", version = "0.1.5" }
diff --git a/gcc-0.3.13/Cargo.toml b/gcc-0.3.13/Cargo.toml
index ab74b23..00ea9af 100644
--- a/gcc-0.3.13/Cargo.toml
+++ b/gcc-0.3.13/Cargo.toml
@@ -12,16 +12,3 @@ C compiler to compile native C code into a static archive to be linked into Rust
 code.
 """
 keywords = ["build-dependencies"]
-
-[target.i686-pc-windows-msvc.dependencies]
-winapi = "0.2.1"
-advapi32-sys = "0.1.2"
-[target.x86_64-pc-windows-msvc.dependencies]
-winapi = "0.2.1"
-advapi32-sys = "0.1.2"
-[target.i686-pc-windows-gnu.dependencies]
-winapi = "0.2.1"
-advapi32-sys = "0.1.2"
-[target.x86_64-pc-windows-gnu.dependencies]
-winapi = "0.2.1"
-advapi32-sys = "0.1.2"
diff --git a/git2-0.2.14/Cargo.toml b/git2-0.2.14/Cargo.toml
index 0de9709..5e25d56 100644
--- a/git2-0.2.14/Cargo.toml
+++ b/git2-0.2.14/Cargo.toml
@@ -16,16 +16,10 @@ repositories.
 """
 
 [dependencies]
-url = "0.2"
-bitflags = "0.1"
-libc = "0.1"
-libgit2-sys = { path = "libgit2-sys", version = "0.2.3" }
-
-[dev-dependencies]
-docopt = "0.6"
-rustc-serialize = "0.3"
-time = "0.1"
-tempdir = "0.3"
+url = { version = "0.2.37", path = "../url-0.2.37" }
+bitflags = { version = "0.1.1", path = "../bitflags-0.1.1" }
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
+libgit2-sys = { path = "../libgit2-sys-0.2.20", version = "0.2.20" }
 
 [features]
 unstable = []
diff --git a/git2-curl-0.2.4/Cargo.toml b/git2-curl-0.2.4/Cargo.toml
index 9f1c71e..d3fcb8f 100644
--- a/git2-curl-0.2.4/Cargo.toml
+++ b/git2-curl-0.2.4/Cargo.toml
@@ -14,19 +14,13 @@ Intended to be used with the git2 crate.
 """
 
 [dependencies]
-curl = "0.2"
-url = "0.2"
-log = "0.3"
+curl = { version = "0.2.11", path = "../curl-0.2.11" }
+url = { version = "0.2.37", path = "../url-0.2.37" }
+log = { version = "0.3.1", path = "../log-0.3.1" }
 
 [dependencies.git2]
-path = ".."
-version = "0.2"
-
-[dev-dependencies]
-civet = "0.7"
-conduit = "0.7"
-conduit-git-http-backend = "0.7"
-tempdir = "0.3"
+path = "../git2-0.2.14"
+version = "0.2.14"
 
 [[test]]
 name = "all"
diff --git a/glob-0.2.10/Cargo.toml b/glob-0.2.10/Cargo.toml
index 504499a..ba123f7 100644
--- a/glob-0.2.10/Cargo.toml
+++ b/glob-0.2.10/Cargo.toml
@@ -10,6 +10,3 @@ documentation = "http://doc.rust-lang.org/glob"
 description = """
 Support for matching file paths against Unix shell style patterns.
 """
-
-[dev-dependencies]
-tempdir = "0.3"
diff --git a/hamcrest-0.1.0/Cargo.toml b/hamcrest-0.1.0/Cargo.toml
index bd5f8da..ad98be3 100644
--- a/hamcrest-0.1.0/Cargo.toml
+++ b/hamcrest-0.1.0/Cargo.toml
@@ -5,4 +5,4 @@ version = "0.1.0"
 authors = ["Carllerche <me@carllerche.com>"]
 
 [dependencies]
-num = "0.1.25"
+num = { version = "0.1.27", path = "../num-0.1.27" }
diff --git a/libgit2-sys-0.2.20/Cargo.toml b/libgit2-sys-0.2.20/Cargo.toml
index e266e63..bb8f778 100644
--- a/libgit2-sys-0.2.20/Cargo.toml
+++ b/libgit2-sys-0.2.20/Cargo.toml
@@ -14,42 +14,42 @@ name = "libgit2_sys"
 path = "lib.rs"
 
 [dependencies]
-libssh2-sys = "0.1.26"
-libc = "0.1"
+libssh2-sys = { version = "0.1.30", path = "../libssh2-sys-0.1.30" }
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
 
 [build-dependencies]
-pkg-config = "0.3"
+pkg-config = { version = "0.3.5", path = "../pkg-config-0.3.5" }
 
 [target.i686-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.x86_64-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.i686-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.x86_64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.aarch64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.arm-unknown-linux-gnueabihf.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.i686-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.x86_64-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.x86_64-unknown-bitrig.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.x86_64-unknown-openbsd.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
 [target.x86_64-unknown-dragonfly.dependencies]
-openssl-sys = "0.6.0"
-libz-sys = "0.1.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
diff --git a/libssh2-sys-0.1.30/Cargo.toml b/libssh2-sys-0.1.30/Cargo.toml
index 3147f2f..0dc13ac 100644
--- a/libssh2-sys-0.1.30/Cargo.toml
+++ b/libssh2-sys-0.1.30/Cargo.toml
@@ -13,32 +13,32 @@ name = "libssh2_sys"
 path = "lib.rs"
 
 [dependencies]
-libz-sys = "0.1.0"
-libc = "0.1"
+libz-sys = { version = "0.1.8", path = "../libz-sys-0.1.8" }
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
 
 [target.i686-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-apple-darwin.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.i686-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.aarch64-unknown-linux-gnu.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.arm-unknown-linux-gnueabihf.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.i686-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-freebsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-dragonfly.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-bitrig.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 [target.x86_64-unknown-openbsd.dependencies]
-openssl-sys = "0.6.0"
+openssl-sys = { version = "0.6.4", path = "../openssl-sys-0.6.4" }
 
 [build-dependencies]
-pkg-config = "0.3"
-cmake = "0.1.2"
+pkg-config = { version = "0.3.5", path = "../pkg-config-0.3.5" }
+cmake = { version = "0.1.3", path = "../cmake-0.1.3" }
diff --git a/libz-sys-0.1.8/Cargo.toml b/libz-sys-0.1.8/Cargo.toml
index d63fd2b..2010cba 100644
--- a/libz-sys-0.1.8/Cargo.toml
+++ b/libz-sys-0.1.8/Cargo.toml
@@ -13,8 +13,8 @@ Bindings to the system libz library (also known as zlib).
 """
 
 [dependencies]
-libc = "0.1"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
 
 [build-dependencies]
-pkg-config = "0.3"
-gcc = "0.3.10"
+pkg-config = { version = "0.3.5", path = "../pkg-config-0.3.5" }
+gcc = { version = "0.3.13", path = "../gcc-0.3.13" }
diff --git a/log-0.3.1/Cargo.toml b/log-0.3.1/Cargo.toml
index 4d61c32..7fb2459 100644
--- a/log-0.3.1/Cargo.toml
+++ b/log-0.3.1/Cargo.toml
@@ -17,4 +17,4 @@ name = "filters"
 harness = false
 
 [dependencies]
-libc = "0.1"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
diff --git a/memchr-0.1.3/Cargo.toml b/memchr-0.1.3/Cargo.toml
index f5f6ec2..3235d06 100644
--- a/memchr-0.1.3/Cargo.toml
+++ b/memchr-0.1.3/Cargo.toml
@@ -14,7 +14,4 @@ license = "Unlicense/MIT"
 name = "memchr"
 
 [dependencies]
-libc = "0.1.*"
-
-[dev-dependencies]
-quickcheck = "0.2.*"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
diff --git a/miniz-sys-0.1.5/Cargo.toml b/miniz-sys-0.1.5/Cargo.toml
index f68115b..c9a0d02 100644
--- a/miniz-sys-0.1.5/Cargo.toml
+++ b/miniz-sys-0.1.5/Cargo.toml
@@ -16,7 +16,7 @@ name = "miniz_sys"
 path = "lib.rs"
 
 [dependencies]
-libc = "0.1"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
 
 [build-dependencies]
-gcc = "0.3"
+gcc = { version = "0.3.13", path = "../gcc-0.3.13" }
diff --git a/num-0.1.27/Cargo.toml b/num-0.1.27/Cargo.toml
index ebe6fca..3dbc60b 100644
--- a/num-0.1.27/Cargo.toml
+++ b/num-0.1.27/Cargo.toml
@@ -14,8 +14,8 @@ rational, and complex types.
 """
 
 [dependencies]
-rustc-serialize = { version = "0.3.13", optional = true }
-rand = { version = "0.3.8", optional = true }
+rustc-serialize = { version = "0.3.16", path = "../rustc-serialize-0.3.16", optional = true }
+rand = { version = "0.3.10", path = "../rand-0.3.10", optional = true }
 
 [features]
 
diff --git a/num_cpus-0.2.6/Cargo.toml b/num_cpus-0.2.6/Cargo.toml
index 298261d..38bd6ee 100644
--- a/num_cpus-0.2.6/Cargo.toml
+++ b/num_cpus-0.2.6/Cargo.toml
@@ -9,4 +9,4 @@ repository = "https://github.com/seanmonstar/num_cpus"
 
 
 [dependencies]
-libc = "0.1.7"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
diff --git a/openssl-sys-0.6.4/Cargo.toml b/openssl-sys-0.6.4/Cargo.toml
index 5ed9d06..a530d27 100644
--- a/openssl-sys-0.6.4/Cargo.toml
+++ b/openssl-sys-0.6.4/Cargo.toml
@@ -22,17 +22,8 @@ npn = []
 alpn = []
 
 [dependencies]
-libc = "0.1"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
 
 [build-dependencies]
-pkg-config = "0.3"
-gcc = "0.3"
-
-[target.le32-unknown-nacl.dependencies]
-libressl-pnacl-sys = "2.1.0"
-[target.x86_64-unknown-nacl.dependencies]
-libressl-pnacl-sys = "2.1.0"
-[target.i686-unknown-nacl.dependencies]
-libressl-pnacl-sys = "2.1.0"
-[target.arm-unknown-nacl.dependencies]
-libressl-pnacl-sys = "2.1.0"
+pkg-config = { version = "0.3.5", path = "../pkg-config-0.3.5" }
+gcc = { version = "0.3.13", path = "../gcc-0.3.13" }
diff --git a/rand-0.3.10/Cargo.toml b/rand-0.3.10/Cargo.toml
index 62e0535..e75db57 100644
--- a/rand-0.3.10/Cargo.toml
+++ b/rand-0.3.10/Cargo.toml
@@ -14,9 +14,4 @@ Random number generators and other randomness functionality.
 keywords = ["random"]
 
 [dependencies]
-libc = "0.1.1"
-winapi = "0.2"
-advapi32-sys = "0.1.2"
-
-[dev-dependencies]
-log = "0.3.0"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
diff --git a/regex-0.1.41/Cargo.toml b/regex-0.1.41/Cargo.toml
index 96b12f9..45bf6f8 100644
--- a/regex-0.1.41/Cargo.toml
+++ b/regex-0.1.41/Cargo.toml
@@ -34,12 +34,9 @@ test = false
 bench = true
 
 [dependencies]
-aho-corasick = "0.3"
-memchr = "0.1"
-regex-syntax = { path = "regex-syntax", version = "0.2" }
-
-[dev-dependencies]
-rand = "0.3"
+aho-corasick = { version = "0.3.0", path = "../aho-corasick-0.3.0" }
+memchr = { version = "0.1.3", path = "../memchr-0.1.3" }
+regex-syntax = { path = "../regex-syntax-0.2.1", version = "0.2.1" }
 
 [features]
 pattern = []
diff --git a/regex-syntax-0.2.1/Cargo.toml b/regex-syntax-0.2.1/Cargo.toml
index bd8f926..41243be 100644
--- a/regex-syntax-0.2.1/Cargo.toml
+++ b/regex-syntax-0.2.1/Cargo.toml
@@ -7,7 +7,3 @@ repository = "https://github.com/rust-lang/regex"
 documentation = "http://doc.rust-lang.org/regex"
 homepage = "https://github.com/rust-lang/regex"
 description = "A regular expression parser."
-
-[dev-dependencies]
-quickcheck = "0.2"
-rand = "0.3"
diff --git a/rustc-serialize-0.3.16/Cargo.toml b/rustc-serialize-0.3.16/Cargo.toml
index d4cc6f7..181ed97 100644
--- a/rustc-serialize-0.3.16/Cargo.toml
+++ b/rustc-serialize-0.3.16/Cargo.toml
@@ -13,6 +13,3 @@ Generic serialization/deserialization support corresponding to the
 `derive(RustcEncodable, RustcDecodable)` mode in the compiler. Also includes
 support for hex, base64, and json encoding and decoding.
 """
-
-[dev-dependencies]
-rand = "0.3"
diff --git a/tar-0.3.1/Cargo.toml b/tar-0.3.1/Cargo.toml
index 88ef166..24d50d1 100644
--- a/tar-0.3.1/Cargo.toml
+++ b/tar-0.3.1/Cargo.toml
@@ -18,8 +18,5 @@ contents are never required to be entirely resident in memory all at once.
 """
 
 [dependencies]
-libc = "0.1"
-filetime = "0.1.5"
-
-[dev-dependencies]
-tempdir = "0.3"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
+filetime = { version = "0.1.5", path = "../filetime-0.1.5" }
diff --git a/tempdir-0.3.4/Cargo.toml b/tempdir-0.3.4/Cargo.toml
index 9c8ca76..de1ee1e 100644
--- a/tempdir-0.3.4/Cargo.toml
+++ b/tempdir-0.3.4/Cargo.toml
@@ -14,4 +14,4 @@ dropped.
 """
 
 [dependencies]
-rand = "0.3"
+rand = { version = "0.3.10", path = "../rand-0.3.10" }
diff --git a/term-0.2.11/Cargo.toml b/term-0.2.11/Cargo.toml
index 3a43167..d5bf307 100644
--- a/term-0.2.11/Cargo.toml
+++ b/term-0.2.11/Cargo.toml
@@ -11,7 +11,3 @@ documentation = "http://doc.rust-lang.org/term"
 description = """
 A terminal formatting library
 """
-
-[dependencies]
-winapi = "0.2"
-kernel32-sys = "0.1"
diff --git a/time-0.1.32/Cargo.toml b/time-0.1.32/Cargo.toml
index 65c0da9..454576a 100644
--- a/time-0.1.32/Cargo.toml
+++ b/time-0.1.32/Cargo.toml
@@ -12,11 +12,5 @@ Utilities for working with time-related functions in Rust.
 """
 
 [dependencies]
-libc = "0.1"
-winapi = "0.2.0"
-kernel32-sys = "0.1.2"
-rustc-serialize = { version = "0.3", optional = true }
-
-[dev-dependencies]
-log = "0.3"
-advapi32-sys = "0.1.2"
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
+rustc-serialize = { version = "0.3.16", path = "../rustc-serialize-0.3.16", optional = true }
diff --git a/toml-0.1.22/Cargo.toml b/toml-0.1.22/Cargo.toml
index d5da252..7dd23bb 100644
--- a/toml-0.1.22/Cargo.toml
+++ b/toml-0.1.22/Cargo.toml
@@ -16,11 +16,8 @@ facilitate deserializing and serializing Rust structures.
 """
 
 [dependencies]
-rustc-serialize = { optional = true, version = "0.3.0" }
+rustc-serialize = { optional = true, version = "0.3.16", path = "../rustc-serialize-0.3.16" }
 serde = { optional = true, version = "0.5" }
 
 [features]
 default = ["rustc-serialize"]
-
-[dev-dependencies]
-rustc-serialize = "0.3"
diff --git a/url-0.2.37/Cargo.toml b/url-0.2.37/Cargo.toml
index 87ad084..9661621 100644
--- a/url-0.2.37/Cargo.toml
+++ b/url-0.2.37/Cargo.toml
@@ -33,5 +33,5 @@ version = "*"
 optional = true
 
 [dependencies]
-rustc-serialize = "0.3"
-matches = "0.1"
+rustc-serialize = { version = "0.3.16", path = "../rustc-serialize-0.3.16" }
+matches = { version = "0.1.2", path = "../matches-0.1.2" }
