diff --git a/meson.build b/meson.build
index d55c4c7..c0ef38e 100644
--- a/meson.build
+++ b/meson.build
@@ -25,7 +25,7 @@ dependencies = [
 ]

 config = configuration_data()
-config.set_quoted('PLUGIN_DIR', join_paths(get_option('prefix'), get_option('datadir'), 'xi-gtk', 'plugins'))
+config.set_quoted('PLUGIN_DIR', join_paths(get_option('prefix'), get_option('datadir'), get_option('datasubdir'), 'plugins'))
 configure_file(configuration: config, output: 'config.h')
 dependencies += meson.get_compiler('vala').find_library('config', dirs: meson.current_source_dir())

diff --git a/meson_options.txt b/meson_options.txt
new file mode 100644
index 0000000..0382e40
--- /dev/null
+++ b/meson_options.txt
@@ -0,0 +1,2 @@
+option('datasubdir', type : 'string', value : 'xi-gtk',
+	description: 'Name of the directory in datadir where plugins should be installed')
--
2.16.4

