diff --git a/src/bootstrap/dist.rs b/src/bootstrap/dist.rs
index 7c463be..eda72ed 100644
--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -313,7 +313,7 @@ pub fn rust_src_location(build: &Build) -> PathBuf {
 pub fn analysis(build: &Build, compiler: &Compiler, target: &str) {
     println!("Dist analysis");
 
-    if build.config.channel != "nightly" {
+    if build.config.channel != "nightly" && build.config.channel != "dev" {
         println!("\tskipping - not on nightly channel");
         return;
     }
diff --git a/src/tools/build-manifest/src/main.rs b/src/tools/build-manifest/src/main.rs
index c2ec421..9dc665e 100644
--- a/src/tools/build-manifest/src/main.rs
+++ b/src/tools/build-manifest/src/main.rs
@@ -224,7 +224,7 @@ impl Builder {
         self.package("rust-docs", &mut manifest.pkg, TARGETS);
         self.package("rust-src", &mut manifest.pkg, &["*"]);
 
-        if self.rust_release == "nightly" {
+        if self.rust_release == "nightly" || self.rust_release == "dev" {
             self.package("rust-analysis", &mut manifest.pkg, TARGETS);
         }
 
@@ -277,7 +277,7 @@ impl Builder {
                         target: target.to_string(),
                     });
                 }
-                if self.rust_release == "nightly" {
+                if self.rust_release == "nightly" || self.rust_release == "dev" {
                     extensions.push(Component {
                         pkg: "rust-analysis".to_string(),
                         target: target.to_string(),
