--- src/libcore/ops.rs
+++ src/libcore/ops.rs
@@ -570,12 +570,6 @@
 
 macro_rules! shl_impl(
     ($($t:ty)*) => ($(
-        #[cfg(stage0)]
-        impl Shl<$t, $t> for $t {
-            #[inline]
-            fn shl(&self, other: &$t) -> $t { (*self) << (*other) }
-        }
-        #[cfg(not(stage0), not(test))]
         impl Shl<$t, $t> for $t {
             #[inline]
             fn shl(&self, other: &$t) -> $t {
@@ -619,12 +613,6 @@
 
 macro_rules! shr_impl(
     ($($t:ty)*) => ($(
-        #[cfg(stage0, not(test))]
-        impl Shr<$t, $t> for $t {
-            #[inline]
-            fn shr(&self, other: &$t) -> $t { (*self) >> (*other) }
-        }
-        #[cfg(not(stage0), not(test))]
         impl Shr<$t, $t> for $t {
             #[inline]
             fn shr(&self, other: &$t) -> $t { (*self) >> (*other as uint) }
